cli/src/main/groovy/NexusPublisher.groovy

```diff
@@ -0,0 +1,62 @@
+import com.sonatype.nexus.api.common.Authentication
+import com.sonatype.nexus.api.common.ServerConfig
+import com.sonatype.nexus.api.repository.v3.DefaultAsset
+import com.sonatype.nexus.api.repository.v3.DefaultComponent
+import com.sonatype.nexus.api.repository.v3.RepositoryManagerV3ClientBuilder
+
+import groovy.cli.commons.CliBuilder
+
+/**
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+@Grab(group='org.slf4j', module='slf4j-simple', version='1.7.25')
+@Grab(group='com.sonatype.nexus', module='nexus-platform-api', version='3.3.20181025-095610.1ce86ad')
+
+cli = new CliBuilder(usage: 'Repository', expandArgumentFiles: true)
```

Comment:
After studying this for a bit it looks like these are all definitions of command line arguments that we want to process and that this happens with the 'cli.parse' command below.  My perception before I started this review was that we were producing and executing a command line, but the opposite appears to be true.  This is essentially like a java main() and we're consuming the command line and then doing some processing on that.

I wonder if an overview sentence or two would be helpful in this case.